File: frontend/src/api/quiz.ts
import axiosInstance from './axios';

type Quiz = {
  quiz_id: number;
  quiz_type: string;
  question: string;
  answer: string;
  option1: string;
  option2: string;
  option3: string;
  option4: string;
  explanations: string;
  answer_explanation: string;
};

type QuizResponse = {
  status: string;
  quizes: Quiz[];
};

type QuizAnswer = {
  quizId: number;
  answer: string;
};

const getTodayQuiz = async (): Promise<QuizResponse> => {
  const response = await axiosInstance.get('/quiz/today');
  console.log('api 실행: ', response.data);
  return response.data;
};

const submitQuizAnswer = async ({
  quizId,
  answer,
}: QuizAnswer): Promise<void> => {
  // console.log(quizId, answer);
  const response = await axiosInstance.post(`/quiz/${quizId}`, {answer});
  return response.data;
};

const getSolvedQuiz = async (): Promise<string[]> => {
  const response = await axiosInstance.get('/quiz');
  // console.log("이거야? ",response.data);
  return response.data;
};

export {getTodayQuiz, submitQuizAnswer, getSolvedQuiz};

File: backend/src/main/java/com/example/donzoom/entity/Quiz.java
package com.example.donzoom.entity;

import com.example.donzoom.constant.QuizType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Quiz {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "quiz_id")
  private Long id;

  @Enumerated(EnumType.STRING)
  private QuizType quizType;

  @Column(columnDefinition = "TEXT")
  private String question;

  private String answer;

  private String option1;
  private String option2;
  private String option3;
  private String option4;

  @Column(columnDefinition = "TEXT")
  private String explanations;
  @Column(columnDefinition = "TEXT")
  private String answerExplanation;


  @Builder
  public Quiz(QuizType quizType, String question, String answer, String option1, String option2,
      String option3, String option4, String explanations, String answerExplanation) {
    this.quizType = quizType;
    this.question = question;
    this.answer = answer;
    this.option1 = option1;
    this.option2 = option2;
    this.option3 = option3;
    this.option4 = option4;
    this.explanations = explanations;
    this.answerExplanation = answerExplanation;
  }

}
